src/Media/CMakeLists.txt

```diff
@@ -10,9 +10,7 @@ set(MEDIA_HEADERS Media.h
                   Audio/AudioPlayer.h
                   Audio/OpenALSoundProvider.h)
 
-if(APPLE)
-  add_definitions(-DFF_API_NEXT -Wno-deprecated-declarations)
-endif()
+add_compile_definitions(FF_API_NEXT)
```

Comment:
maybe better to use `target_compile_definitions(media PRIVATE FF_API_NEXT)` after `add_library(media ...)`?